### DISCUSSION TITLE
Didnt work dependency_overrides

### DISCUSSION THREAD

**Discussion by KenKi0** [fastapi/fastapi#8608] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.get("/", response_model=List[RoomPayload])
def read(db: Session = Depends(get_db),
               user=Depends(manager)):

    q = db.query(Room).all()

    if not q:
        raise HTTPException(status_code=404, detail=f"Rooms not found")

    return q

@pytest.fixture(scope="session")
def db_engine():
    engine = create_engine(SQLALCHEMY_DATABASE_URL)
    if not database_exists:
        create_database(engine.url)

    Base.metadata.create_all(bind=engine)

    yield engine


@pytest.fixture(scope="function")
def db(db_engine):
    connection = db_engine.connect()
    connection.begin()
    db = Session(bind=connection)

    yield db

    db.rollback()
    connection.close()


@pytest.fixture(scope="function")
def client(db):
    app.dependency_overrides[get_db] = lambda: db

    with TestClient(app) as c:
        yield c
```

### Description

I am trying to write test for web service and I want to create a separate database for tests when you run them. It is my pytest fixture for realise it and on of my endpoints. But app.dependecy_overrides[get_db] = lambda: db didnt work and requests continue to be sent to the main database and not the test one.

### Operating System

Windows

### FastAPI Version

fastapi==0.66

### Python Version

3.8

**Comment by SaraFarron** [comment]
same on fastapi==0.79

**Reply by Sube-py** [reply]
same on fastapi==0.94.1, did you solve it？？？

**Comment by Kludex** [comment]
Please share a minimal, reproducible, example. 🙏
